nil/internal/execution/block_generator.go

```diff
@@ -153,6 +153,11 @@ func (p *BlockGenerator) CollectGasPrices(prevBlockId types.BlockNumber) []types
 			shardHash = mainBlock.PrevBlock
 		}
 
+		if shardHash.Empty() {
+			shards[shardId] = *types.DefaultGasPrice.Uint256
```

Comment:
You can fill the dictionary with default values at once, then assignment in two branches of the default value will become unnecessary